### DISCUSSION TITLE
Printing to local printer

### DISCUSSION THREAD

**Discussion by andrewbonsey** [sumatrapdfreader/sumatrapdf#4060] [Q&A]
I have this script that throws an error about the local printer not being a recognized name for the cmdlet and also that the PDF fails to scale when output. Any help of why the code is wrong is greatly appreciated.

# Declare variables
$pdfFolder = "C:\HF" # Replace with the actual folder path
$printerName = ZDesignerGK420d # Replace with the correct printer name

# Get all PDF files in the folder
$pdfFiles = Get-ChildItem -Path $pdfFolder -Filter "*.pdf"

# Check if any PDF files were found
if ($pdfFiles.Count -eq 0) {
  Write-Host "No PDF files found in the specified folder."
  Exit
}

# Print each PDF file using SumatraPDF
foreach ($pdfFile in $pdfFiles) {
  # Use Start-Process for reliable cross-platform printing
  Start-Process -FilePath "C:\Users\andre\AppData\Local\SumatraPDF\SumatraPDF.exe" -ArgumentList "-print-settings fit", "-print-to ZDesignerGK420d",  $pdfFile.FullName  -Wait
}

Write-Host "Successfully printed $($pdfFiles.Count) PDF files."

**Comment by GitHubRulesOK** [comment]
The command line interface is designed for the command line so unsupported if using from inside another method.

debug by write a valid command line in the console then any changes around that are yours.

error messages are usually clear "local printer not being a recognized name for the cmdlet " 
suggests your attempting to run a printer as a command 
rather than call SumatraPDF.
